# Merging DataFrames for EDA#

In [1]:
import pandas as pd

In [2]:
!pwd

/Users/italo/Desktop/Course Material/Module1 Project/2020-movie-analysis


## Import Movie Basics and Ratings From INDB ##

In [79]:
basics_and_ratings = pd.read_csv('Data/imdb.basics_and_ratings_nonulls.csv')
print(basics_and_ratings.shape)
basics_and_ratings.head()

(73856, 8)


title_id                               primary_title  \
0  tt1612608                                 The Italian   
1  tt4874854  The Legend of Zelda: The Triforce Prophecy   
2  tt1718842                          Nakalnya anak muda   
3  tt1645134                                  Sandcastle   
4  tt1645129                                     Planzet   

                               original_title  start_year  runtime_minutes  \
0                                   L'Italien        2010            102.0   
1  The Legend of Zelda: The Triforce Prophecy        2010            120.0   
2                          Nakalnya anak muda        2010             75.0   
3                                  Sandcastle        2010             91.0   
4                                  Puranzetto        2010             53.0   

                      genres  averagerating  numvotes  
0               Comedy,Drama            5.8      1360  
1  Adventure,Fantasy,Romance            4.8         8  
2            Horror,Thriller            4.0        12  
3                      Drama            6.3       116  
4    Action,Animation,Sci-Fi            4.6       290

In [80]:
basics_and_ratings.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 73856 entries, 0 to 73855
Data columns (total 8 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   title_id         73856 non-null  object 
 1   primary_title    73856 non-null  object 
 2   original_title   73856 non-null  object 
 3   start_year       73856 non-null  int64  
 4   runtime_minutes  66236 non-null  float64
 5   genres           73052 non-null  object 
 6   averagerating    73856 non-null  float64
 7   numvotes         73856 non-null  int64  
dtypes: float64(2), int64(2), object(4)
memory usage: 4.5+ MB


In [81]:
basics_and_ratings.original_title.nunique()

# there are 73856 unique entries in this table. But there are only  69993 unique primary titles, this means that
# different movies share the same primary title. Only 71097 original titles, as well. 

71097

In [82]:
# information from 2010 - 2019
basics_and_ratings.start_year.value_counts()

2016    8721
2017    8713
2015    8494
2014    8371
2013    7990
2012    7680
2018    7526
2011    7389
2010    6792
2019    2180
Name: start_year, dtype: int64

### clean up null values ###

In [83]:
basics_and_ratings.isnull().sum()

# runtime has a variety of null values, but our questions don't require runtime so best to drop the column

title_id              0
primary_title         0
original_title        0
start_year            0
runtime_minutes    7620
genres              804
averagerating         0
numvotes              0
dtype: int64

In [84]:
b_and_r_cleaned = basics_and_ratings.drop('runtime_minutes', axis = 1)
b_and_r_cleaned

title_id                               primary_title  \
0       tt1612608                                 The Italian   
1       tt4874854  The Legend of Zelda: The Triforce Prophecy   
2       tt1718842                          Nakalnya anak muda   
3       tt1645134                                  Sandcastle   
4       tt1645129                                     Planzet   
...           ...                                         ...   
73851  tt10078894                           Deadly Playthings   
73852   tt6643972                               Triple Threat   
73853   tt7936092     Black Snake: La légende du serpent noir   
73854  tt10080924                                     Sweater   
73855   tt9916160                                  Drømmeland   

                                   original_title  start_year  \
0                                       L'Italien        2010   
1      The Legend of Zelda: The Triforce Prophecy        2010   
2                              Nakalnya anak muda        2010   
3                                      Sandcastle        2010   
4                                      Puranzetto        2010   
...                                           ...         ...   
73851                           Deadly Playthings        2019   
73852                               Triple Threat        2019   
73853     Black Snake: La légende du serpent noir        2019   
73854                                     Sweater        2019   
73855                                  Drømmeland        2019   

                          genres  averagerating  numvotes  
0                   Comedy,Drama            5.8      1360  
1      Adventure,Fantasy,Romance            4.8         8  
2                Horror,Thriller            4.0        12  
3                          Drama            6.3       116  
4        Action,Animation,Sci-Fi            4.6       290  
...                          ...            ...       ...  
73851                     Horror            2.3        30  
73852            Action,Thriller            5.5      4812  
73853                     Comedy            4.9        62  
73854                      Drama            9.2        79  
73855                Documentary            6.5        11  

[73856 rows x 7 columns]

In [88]:
# still need to figure out what to do null values for genres. seems that they are foreign films, do we want to keep
# these? or drop them?
b_and_r_cleaned[b_and_r_cleaned.genres.isnull()]

title_id                                primary_title  \
32      tt1645166                                       Torino   
240     tt1649393                                      Piedras   
333     tt1621833                           My American Exodus   
342     tt3678082  Min pinlige familie og mutantdræbersneglene   
374     tt2007453                                       Sailor   
...           ...                                          ...   
72559   tt9251858                                        Hippi   
72837  tt10161980                             Ituzaingo V3rit4   
72887  tt10161248                            Método Livingston   
73637  tt10122528                                   Miel-Emile   
73755   tt7907676                                    Alexandra   

                                    original_title  year genres  \
32                                          Torino  2010    NaN   
240                                        Piedras  2010    NaN   
333                             My American Exodus  2010    NaN   
342    Min pinlige familie og mutantdræbersneglene  2010    NaN   
374                                         Sailor  2010    NaN   
...                                            ...   ...    ...   
72559                                        Hippi  2019    NaN   
72837                             Ituzaingo V3rit4  2019    NaN   
72887                            Método Livingston  2019    NaN   
73637                                   Miel-Emile  2019    NaN   
73755                                   Aleksandra  2019    NaN   

       averagerating  numvotes  
32               7.1        12  
240              4.9         7  
333              7.6         5  
342              8.5         6  
374              7.6        17  
...              ...       ...  
72559            8.2        28  
72837            4.2         5  
72887            8.2         5  
73637            8.2         5  
73755            6.7         7  

[804 rows x 7 columns]

### change start_year to year ###

In [86]:
b_and_r_cleaned.rename(columns = {'start_year' : 'year'}, inplace = True)
print(b_and_r_cleaned.shape)
b_and_r_cleaned.head()

(73856, 7)


title_id                               primary_title  \
0  tt1612608                                 The Italian   
1  tt4874854  The Legend of Zelda: The Triforce Prophecy   
2  tt1718842                          Nakalnya anak muda   
3  tt1645134                                  Sandcastle   
4  tt1645129                                     Planzet   

                               original_title  year  \
0                                   L'Italien  2010   
1  The Legend of Zelda: The Triforce Prophecy  2010   
2                          Nakalnya anak muda  2010   
3                                  Sandcastle  2010   
4                                  Puranzetto  2010   

                      genres  averagerating  numvotes  
0               Comedy,Drama            5.8      1360  
1  Adventure,Fantasy,Romance            4.8         8  
2            Horror,Thriller            4.0        12  
3                      Drama            6.3       116  
4    Action,Animation,Sci-Fi            4.6       290

## Import Box Office Mojo Data Set ##

In [91]:
bom_gross = pd.read_csv('Data/bom.movie_gross.csv', thousands = ',') # thousands parameter gets rid of commas 
print(bom_gross.shape)
bom_gross.head()

(3387, 5)


title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

   foreign_gross  year  
0    652000000.0  2010  
1    691300000.0  2010  
2    664300000.0  2010  
3    535700000.0  2010  
4    513900000.0  2010

In [92]:
# information from years 2010 - 2018
bom_gross.year.value_counts()

2015    450
2016    436
2012    400
2011    399
2014    395
2013    350
2010    328
2017    321
2018    308
Name: year, dtype: int64

### clean null values ###

In [93]:
# check for null values 
bom_gross.isnull().sum()

title                0
studio               5
domestic_gross      28
foreign_gross     1350
year                 0
dtype: int64

In [94]:
# fill nulls in domestic and foreign gross with 0's
bom_gross.domestic_gross = bom_gross.domestic_gross.fillna(0)
bom_gross.foreign_gross = bom_gross.foreign_gross.fillna(0)
bom_gross.isnull().sum()

title             0
studio            5
domestic_gross    0
foreign_gross     0
year              0
dtype: int64

In [95]:
# check null values for studio, since there is only 5 with null values just drop those rows
bom_gross[bom_gross.studio.isnull()]

title studio  domestic_gross  foreign_gross  \
210   Outside the Law (Hors-la-loi)    NaN         96900.0      3300000.0   
555         Fireflies in the Garden    NaN         70600.0      3300000.0   
933           Keith Lemon: The Film    NaN             0.0      4000000.0   
1862                 Plot for Peace    NaN          7100.0            0.0   
2825               Secret Superstar    NaN             0.0    122000000.0   

      year  
210   2010  
555   2011  
933   2012  
1862  2014  
2825  2017

In [96]:
bom_gross = bom_gross.dropna(axis = 0, subset = ['studio'])

In [97]:
# successfully dropped null values
bom_gross.isnull().sum()

title             0
studio            0
domestic_gross    0
foreign_gross     0
year              0
dtype: int64

 ### make new column called total_gross that is sum of domestic gross and foreign gross ###


In [98]:
print(bom_gross.domestic_gross.dtype)
print(bom_gross.foreign_gross.dtype)

float64
float64


In [99]:
# make new column called total_gross that is sum of domestic gross and foreign gross
total_gross = bom_gross['domestic_gross'] + bom_gross['foreign_gross']
bom_gross['total_gross'] = total_gross

In [100]:
# would like to figure out a way to change the scientific notation to standard form
bom_gross_cleaned = bom_gross
print(bom_gross_cleaned.shape)
bom_gross_cleaned.head()

(3382, 6)


title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

   foreign_gross  year   total_gross  
0    652000000.0  2010  1.067000e+09  
1    691300000.0  2010  1.025500e+09  
2    664300000.0  2010  9.603000e+08  
3    535700000.0  2010  8.283000e+08  
4    513900000.0  2010  7.526000e+08

In [145]:
bom_gross_cleaned.title.nunique()

3381

## Import The Numbers Data Set ##

In [29]:
budgets = pd.read_csv('Data/tn.movie_budgets.csv')
print(budgets.shape)
budgets.head()

(5782, 6)


id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [30]:
budgets.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


### clean up release_date column ###

In [51]:
# extract year and convert it to int64 for one case
example = budgets.release_date[0]
int(example[-4:])

int

In [54]:
# use lambda function to apply to entire column
budgets.release_date = budgets.release_date.apply(lambda x : int(x[-4:]))

In [55]:
budgets.head()

id  release_date                                        movie  \
0   1          2009                                       Avatar   
1   2          2011  Pirates of the Caribbean: On Stranger Tides   
2   3          2019                                 Dark Phoenix   
3   4          2015                      Avengers: Age of Ultron   
4   5          2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [58]:
# rename release date to year, for consistency across tables 
budgets.rename(columns = {'release_date' : 'year'}, inplace = True)

In [60]:
budgets.head()

id  year                                        movie production_budget  \
0   1  2009                                       Avatar      $425,000,000   
1   2  2011  Pirates of the Caribbean: On Stranger Tides      $410,600,000   
2   3  2019                                 Dark Phoenix      $350,000,000   
3   4  2015                      Avengers: Age of Ultron      $330,600,000   
4   5  2017            Star Wars Ep. VIII: The Last Jedi      $317,000,000   

  domestic_gross worldwide_gross  
0   $760,507,625  $2,776,345,279  
1   $241,063,875  $1,045,663,875  
2    $42,762,350    $149,762,350  
3   $459,005,868  $1,403,013,963  
4   $620,181,382  $1,316,721,747

In [62]:
# sort by dataframe by year while we are at it
budgets.sort_values(by = 'year', inplace  = True)
budgets.head()

id  year                           movie production_budget  \
5677  78  1915           The Birth of a Nation          $110,000   
5523  24  1916                     Intolerance          $385,907   
5614  15  1916    20,000 Leagues Under the Sea          $200,000   
5683  84  1920  Over the Hill to the Poorhouse          $100,000   
4569  70  1925   Ben-Hur: A Tale of the Christ        $3,900,000   

     domestic_gross worldwide_gross  
5677    $10,000,000     $11,000,000  
5523             $0              $0  
5614     $8,000,000      $8,000,000  
5683     $3,000,000      $3,000,000  
4569     $9,000,000      $9,000,000

In [119]:
# return dataframe that only has movie data from 2000 - 2020
budgets_cleaned = budgets[budgets.year >= 2000]
print(budgets_cleaned.shape)
budgets_cleaned.head()

(4387, 6)


id  year             movie production_budget domestic_gross  \
5288  89  2000      The Specials        $1,000,000        $12,996   
3308   9  2000         The Claim       $13,000,000       $622,023   
1863  64  2000  Rugrats in Paris       $30,000,000    $76,501,438   
324   25  2000       The Patriot      $110,000,000   $113,330,342   
3891  92  2000   American Psycho        $8,000,000    $15,070,285   

     worldwide_gross  
5288         $12,996  
3308      $1,375,635  
1863    $103,284,813  
324     $215,300,000  
3891     $34,270,285

### convert production_budget, domest_gross, worldwide_gross to int ##

In [120]:
# extract string, get rid of comma and dollar sign, convert to integer
example = budgets_cleaned.production_budget[0]
int(example.replace('$', '').replace(',', ''))

425000000

In [121]:
# define function 
def string_to_int(num_str) :
    return int(num_str.replace('$', '').replace(',', ''))

In [124]:
# apply function to all entries in a column
budgets_cleaned.production_budget = budgets_cleaned.production_budget.apply(string_to_int)

/opt/anaconda3/lib/python3.8/site-packages/pandas/core/generic.py:5303: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self[name] = value


In [128]:
# apply function to remaining columns
budgets_cleaned.domestic_gross = budgets_cleaned.domestic_gross.apply(string_to_int)
budgets_cleaned.worldwide_gross = budgets_cleaned.worldwide_gross.apply(string_to_int)

/opt/anaconda3/lib/python3.8/site-packages/pandas/core/generic.py:5303: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self[name] = value


In [129]:
budgets_cleaned.head()

id  year             movie  production_budget  domestic_gross  \
5288  89  2000      The Specials            1000000           12996   
3308   9  2000         The Claim           13000000          622023   
1863  64  2000  Rugrats in Paris           30000000        76501438   
324   25  2000       The Patriot          110000000       113330342   
3891  92  2000   American Psycho            8000000        15070285   

      worldwide_gross  
5288            12996  
3308          1375635  
1863        103284813  
324         215300000  
3891         34270285

In [135]:
# rename worldwide_gross to total gross for consistency
budgets_cleaned.rename(columns = {'worldwide_gross' : 'total_gross'}, inplace = True)
budgets_cleaned.head()

/opt/anaconda3/lib/python3.8/site-packages/pandas/core/frame.py:4125: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().rename(


id  year             movie  production_budget  domestic_gross  \
5288  89  2000      The Specials            1000000           12996   
3308   9  2000         The Claim           13000000          622023   
1863  64  2000  Rugrats in Paris           30000000        76501438   
324   25  2000       The Patriot          110000000       113330342   
3891  92  2000   American Psycho            8000000        15070285   

      total_gross  
5288        12996  
3308      1375635  
1863    103284813  
324     215300000  
3891     34270285

### examine null values in the data frame ###

In [133]:
# no null values in the data frame :)
budgets_cleaned.isnull().any()

id                   False
year                 False
movie                False
production_budget    False
domestic_gross       False
worldwide_gross      False
dtype: bool

### determining movie profits ###

In [136]:
# determine profitability for each movie as ratio of total_gross / production_budget
budgets_cleaned['profitability'] = budgets_cleaned['total_gross'] / budgets_cleaned['production_budget']

<ipython-input-136-e71febc83e22>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  budgets_cleaned['profitability'] = budgets_cleaned['total_gross'] / budgets_cleaned['production_budget']


In [147]:
print(budgets_cleaned.shape)
budgets_cleaned.head()

(4387, 7)


id  year             movie  production_budget  domestic_gross  \
5288  89  2000      The Specials            1000000           12996   
3308   9  2000         The Claim           13000000          622023   
1863  64  2000  Rugrats in Paris           30000000        76501438   
324   25  2000       The Patriot          110000000       113330342   
3891  92  2000   American Psycho            8000000        15070285   

      total_gross  profitability  
5288        12996       0.012996  
3308      1375635       0.105818  
1863    103284813       3.442827  
324     215300000       1.957273  
3891     34270285       4.283786

In [138]:
budgets_cleaned.sort_values(by = 'profitability', ascending = False)

id  year                     movie  production_budget  domestic_gross  \
5492   93  2009       Paranormal Activity             450000       107918810   
5679   80  2015               The Gallows             100000        22764410   
5709   10  2004             Super Size Me              65000        11529368   
5781   82  2005         My Date With Drew               1100          181041   
5656   57  2007                      Once             150000         9445857   
...   ...   ...                       ...                ...             ...   
4850   51  2014         Not Safe For Work            2500000               0   
5599  100  2014                Horse Camp             250000               0   
5565   66  2015  Amidst the Devil's Wings             300000               0   
5640   41  2015            Perfect Cowboy             200000               0   
535    36  2020          Call of the Wild           82000000               0   

      total_gross  profitability  
5492    194183034     431.517853  
5679     41656474     416.564740  
5709     22233808     342.058585  
5781       181041     164.582727  
5656     23323631     155.490873  
...           ...            ...  
4850            0       0.000000  
5599            0       0.000000  
5565            0       0.000000  
5640            0       0.000000  
535             0       0.000000  

[4387 rows x 7 columns]

In [139]:
budgets_cleaned.sort_values(by = 'production_budget', ascending = False)

id  year                                        movie  \
0      1  2009                                       Avatar   
1      2  2011  Pirates of the Caribbean: On Stranger Tides   
2      3  2019                                 Dark Phoenix   
3      4  2015                      Avengers: Age of Ultron   
4      5  2017            Star Wars Ep. VIII: The Last Jedi   
...   ..   ...                                          ...   
5774  75  2004                                       Primer   
5776  77  2004                              The Mongol King   
5779  80  2005                Return to the Land of Wonders   
5780  81  2015                         A Plague So Pleasant   
5781  82  2005                            My Date With Drew   

      production_budget  domestic_gross  total_gross  profitability  
0             425000000       760507625   2776345279       6.532577  
1             410600000       241063875   1045663875       2.546673  
2             350000000        42762350    149762350       0.427892  
3             330600000       459005868   1403013963       4.243841  
4             317000000       620181382   1316721747       4.153696  
...                 ...             ...          ...            ...  
5774               7000          424760       841926     120.275143  
5776               7000             900          900       0.128571  
5779               5000            1338         1338       0.267600  
5780               1400               0            0       0.000000  
5781               1100          181041       181041     164.582727  

[4387 rows x 7 columns]

In [143]:
budgets_cleaned.sort_values(by = 'profitability')

id  year                movie  production_budget  domestic_gross  \
535   36  2020     Call of the Wild           82000000               0   
4565  66  2012  How to Fall in Love            4000000               0   
5330  31  2012               Trance             950000               0   
5311  12  2012         Truth or Die            1000000               0   
5427  28  2012            High Road             546173               0   
...   ..   ...                  ...                ...             ...   
5656  57  2007                 Once             150000         9445857   
5781  82  2005    My Date With Drew               1100          181041   
5709  10  2004        Super Size Me              65000        11529368   
5679  80  2015          The Gallows             100000        22764410   
5492  93  2009  Paranormal Activity             450000       107918810   

      total_gross  profitability  
535             0       0.000000  
4565            0       0.000000  
5330            0       0.000000  
5311            0       0.000000  
5427            0       0.000000  
...           ...            ...  
5656     23323631     155.490873  
5781       181041     164.582727  
5709     22233808     342.058585  
5679     41656474     416.564740  
5492    194183034     431.517853  

[4387 rows x 7 columns]

In [144]:
budgets_cleaned[budgets_cleaned.profitability != 0].sort_values(by = 'profitability')

id  year                movie  production_budget  domestic_gross  \
5302   3  2015        American Hero            1000000               0   
4081  82  2010   Perrier’s Bounty            6600000             828   
3818  19  2015           Skin Trade            9000000            1242   
5335  36  2014          Food Chains             913000               0   
3161  62  2015        Broken Horses           15000000               0   
...   ..   ...                  ...                ...             ...   
5656  57  2007                 Once             150000         9445857   
5781  82  2005    My Date With Drew               1100          181041   
5709  10  2004        Super Size Me              65000        11529368   
5679  80  2015          The Gallows             100000        22764410   
5492  93  2009  Paranormal Activity             450000       107918810   

      total_gross  profitability  
5302           26       0.000026  
4081          828       0.000125  
3818         1242       0.000138  
5335          176       0.000193  
3161         3471       0.000231  
...           ...            ...  
5656     23323631     155.490873  
5781       181041     164.582727  
5709     22233808     342.058585  
5679     41656474     416.564740  
5492    194183034     431.517853  

[4045 rows x 7 columns]

## Export Cleaned Data to New Folder ##

In [148]:
!pwd

/Users/italo/Desktop/Course Material/Module1 Project/2020-movie-analysis


In [149]:
b_and_r_cleaned.to_csv('Cleaned Data/cleaned_imdb-basics-and-ratings.csv')
bom_gross_cleaned.to_csv('Cleaned Data/cleaned_bom-movie-gross.csv')
budgets_cleaned.to_csv('Cleaned Data/cleaned_tn-movie-budgets.csv')

## Cleaned Data Sets Summary ##

**b_and_r_cleaned** gives title, year, genre, ratings, and number of votes for each movie. It has unique 73857 entries though there are many duplicate movie titles in the data frame, i.e. movies that share the same name. This data set can be used to for a ratings analysis for individual movies. Movies are from 2010 - 2019.

**bom_gross_cleaned** gives title, production studio, year, domestic gross, foreign gross, and total gross for each movie. There are 3382 entries in the data frame though only 3381 titles are unique (could be a duplicate?). This data set is good for analyzing yearly profits aggregated by studio. Can also be used to determine a studios yearly market share and trends over time. Can be used to determine profits over time, e.g. Movies are from 2010 - 2018.

**budgets_cleaned** gives year, title, production budget, domestic gross, total gross, and profitability for each movie. It has 4387 entries in the data frame. However, there are only 4364 unique movie titles implying that either movies have the same name or are duplicates. It can be used to see correlation between production budget and total gross or production budget and profitability. Movies are from 2000 - 2020. As such, this data set can be used in our analysis of 2008 recession and the correlation between unemployment rate and profitability of movies. 